chore: add csv example (#4904)
apache/arrow-rs: arrow-csv/examples/README.md

<!---
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->

# Examples

> - [`csv_calculation.rs`](csv_calculation.rs): simple use the csv API to calcuate.

Reviewer: ```suggestion
- [`csv_calculation.rs`](csv_calculation.rs): performs a simple calculation using the CSV reader
```
